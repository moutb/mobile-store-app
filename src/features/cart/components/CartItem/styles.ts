import { media } from '@/styles/media';
import Image from 'next/image';
import styled from 'styled-components';
import { css } from 'styled-components';

export const ItemContainer = styled.div`
    display: flex;
    gap: var(--spacing-4);
    align-items: center;
    padding-bottom: var(--spacing-4);
`;

export const ProductImageWrapper = styled.div`
    position: relative;
    height: var(--product-image-wrapper-height);
    width: 40%;

    ${media('md')(css`
        height: var(--product-image-wrapper-md-width);
        width: var(--product-image-wrapper-md-width);
    `)}
`;

export const ProductImage = styled(Image)`
    flex-shrink: 0;
`;

export const Info = styled.div`
    flex: 1;
    display: flex;
    flex-direction: column;
    gap: var(--spacing-2);
    height: 100%;
    max-height: var(--product-info-max-height);
`;

const ProductTextInfo = css`
    font-size: var(--font-size-sm);
    font-weight: var(--font-weight-thin);
    text-transform: uppercase;
`;

export const Name = styled.h2`
    ${ProductTextInfo}
`;

export const Description = styled.p`
    ${ProductTextInfo}
`;

export const Price = styled.p`
    ${ProductTextInfo}
    margin-top: var(--spacing-4);
`;

export const Spacer = styled.div`
    flex-grow: 1;
    flex-shrink: 0;
    width: 1px;
`;

export const RemoveButton = styled.button`
    align-self: flex-start;
    background: none;
    border: none;
    padding: 0;
    color: var(--color-danger);
    font-size: var(--font-size-sm);
    cursor: pointer;

    &:hover {
        text-decoration: underline;
    }

    &:focus-visible {
        outline: 2px solid var(--color-primary);
        outline-offset: 2px;
    }
`;
