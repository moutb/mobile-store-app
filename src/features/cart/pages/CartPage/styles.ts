'use client';

import styled from 'styled-components';

export const CartContainer = styled.section`
    /* 100vh - header */
    height: calc(100vh - var(--header-height));
    display: flex;
    flex-direction: column;
    padding-top: var(--spacing-6);
    gap: var(spacing-4);
    justify-content: stretch;
`;

export const CartTitle = styled.h1`
    font-size: var(--font-size-lg);
    font-weight: var(--font-weight-thin);
`;

export const EmptyMessage = styled.p`
    font-size: var(--font-size-base);
    text-align: center;
    margin-top: var(--spacing-6);
`;

export const CartItemsContainer = styled.div`
    display: flex;
    margin-top: var(--spacing-6);
    width: 100%;
    justify-content: space-between;
    padding-bottom: var(--spacing-6);
    flex: 1;
    flex-flow: column nowrap;

    & > * {
        border-bottom: 0.5px solid var(--color-primary);
    }
    & > *:nth-last-child(2),
    & > *:last-child {
        border-bottom: none;
    }
`;
