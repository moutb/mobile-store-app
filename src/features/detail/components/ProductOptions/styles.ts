import styled from 'styled-components';

export const OptionsContainer = styled.fieldset`
    border: none;
    width: 100%;
`;

export const OptionTitle = styled.legend`
    text-transform: uppercase;
    font-size: var(--product-option-title-font-size);
    font-weight: var(--product-option-title-font-weight);
    line-height: var(--product-option-title-line-height);
    margin-bottom: var(--product-option-title-margin-bottom);
    letter-spacing: var(--product-option-title-letter-spacing);
`;

export const ColorOptions = styled.div`
    display: flex;
    gap: var(--product-color-options-gap);
    margin-bottom: 1rem;
`;

export const ColorBox = styled.div<{ $selected: boolean }>`
    width: var(--product-color-options-width);
    height: var(--product-color-options-height);
    transition: border 0.2s ease-in-out;
    border-width: ${({ $selected }) => ($selected ? '2px' : '1px')};
    border-style: solid;
    border-color: ${({ theme, $selected }) =>
        $selected ? theme.colors.primary : theme.colors.light};
    box-shadow: inset 0 0 0 1px white;
    cursor: pointer;

    &:focus-visible {
        outline: 2px var(--color-focus);
        outline-offset: 2px;
        border-radius: 4px;
    }
`;

export const ColorName = styled.span`
    font-size: var(--product-color-option-name-font-size);
    font-weight: var(--product-color-option-name-font-weight);
`;

export const StorageGroup = styled.div`
    display: flex;
    flex-wrap: wrap;
`;

export const StorageOption = styled.div<{ $selected: boolean }>`
    padding: var(--product-storage-options-padding);
    transition: border 0.6s ease-in-out;
    border: 1px solid
        ${({ theme, $selected }) =>
            $selected ? theme.colors.primary : theme.colors.light};
    background: ${({ $selected }) => ($selected ? '#fff' : 'transparent')};
    cursor: pointer;
    font: inherit;
    font-weight: var(--product-storage-options-font-weight);

    &:focus-visible {
        outline: 2px var(--color-focus);
        outline-offset: 2px;
        border-radius: 4px;
    }
`;
