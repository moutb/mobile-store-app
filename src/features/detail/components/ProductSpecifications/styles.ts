import styled, { css } from 'styled-components';

export const Specs = styled.section`
    grid-column: 1 / -1;
    margin-top: ${({ theme }) => theme.spacing(6)};
`;

export const SpecsTitle = styled.h2`
    text-transform: uppercase;
    font-size: var(--product-specs-table-title-font-size);
    font-weight: var(--product-specs-table-title-font-weight);
`;

export const SpecsTable = styled.table`
    width: 100%;
    border-collapse: collapse;
    margin-top: var(--product-specs-table-margin-top);
`;

export const SpecsRow = styled.tr`
    border-top: var(--product-specs-table-border-width) solid var(--product-specs-table-border-color);
    border-bottom: var(--product-specs-table-border-width) solid var(--product-specs-table-border-color););
`;

const cellStyle = css`
    font-weight: var(--product-specs-table-cell-font-weight);
    line-height: var(--product-specs-table-cell-line-height);
    font-size: var(--product-specs-table-cell-font-size);
    padding: var(--product-specs-table-cell-padding);
`;

export const SpecsHeader = styled.th`
    text-align: left;
    text-transform: uppercase;
    width: 40%;
    vertical-align: top;
    ${cellStyle}
`;

export const SpecsData = styled.td`
    ${cellStyle}
`;
