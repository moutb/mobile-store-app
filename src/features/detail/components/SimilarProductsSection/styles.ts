import styled from 'styled-components';

export const Wrapper = styled.div`
    margin-top: ${({ theme }) => theme.spacing(4)};
    width: var(--font-siedth);
`;

export const SectionTitle = styled.h2`
    font-size: 1.25rem;
    margin-bottom: ${({ theme }) => theme.spacing(2)};
`;

export const Grid = styled.div`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(140px, 1fr));
    gap: ${({ theme }) => theme.spacing(2)};
`;
