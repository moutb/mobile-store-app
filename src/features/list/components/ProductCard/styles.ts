import Image from 'next/image';
import styled from 'styled-components';

export const Card = styled.article`
    padding: var(--spacing-4);
    border: var(--product-card-border-width) solid var(--color-primary);
    position: relative;
    background: var(--color-background);
    transition: background-color 0.3s ease-in-out;

    &::before {
        content: '';
        position: absolute;
        bottom: 0;
        left: 0;
        width: 100%;
        height: 0%;
        background-color: var(--color-primary);
        z-index: 0;
        transition: height 0.8s cubic-bezier(0.1, 1, 0.8, 1);
    }

    &:hover::before {
        height: 100%;
    }

    & > * {
        position: relative;
        z-index: 1;
    }
`;

export const ImageWrapper = styled.div`
    width: 100%;
    aspect-ratio: 1 / 1;
    display: flex;
    justify-content: center;
    align-items: center;
    overflow: hidden;
`;

export const ProductImage = styled(Image)`
    width: 100%;
    height: 100%;
    object-fit: contain;
    user-drag: none;
    -webkit-user-drag: none;
    pointer-events: auto;
`;

export const ProductLink = styled.a`
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    height: 100%;
    text-decoration: none;
    color: inherit;
    gap: 1rem;
    transition: color 0.3s ease-in-out;

    ${Card}:hover & {
        color: var(--color-white);
    }
`;

export const Info = styled.div`
    display: grid;
    grid-template-columns: 1fr auto;
    gap: var(--spacing-1);
`;

export const Brand = styled.span`
    grid-column: 1 / -1;
    font-size: var(--product-card-brand-font-size);
    text-transform: uppercase;
    font-weight: var(--product-card-brand-font-weight);
`;

export const Name = styled.h2`
    font-size: var(--product-card-name-font-size);
    text-transform: uppercase;
    font-weight: var(--product-card-name-font-weight);
    line-height: var(--product-card-name-line-height);
`;

export const Price = styled.span`
    font-size: var(--product-card-price-font-size);
    font-weight: var(--product-card-price-font-weight);
    line-height: var(--product-card-price-line-height);
    align-self: end;
    justify-self: end;
`;
