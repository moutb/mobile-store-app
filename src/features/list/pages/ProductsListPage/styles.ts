import styled from 'styled-components';

export const ListWrapper = styled.div`
    display: grid;
    grid-template-columns: repeat(
        auto-fit,
        minmax(var(--product-card-width), 1fr)
    );
    width: 100%;
    margin-top: var(--spacing-4);

    & > * {
        border: var(--product-card-border-width) solid var(--color-primary);
        height: 344px;
    }

    --product-card-border-width: 1px;
    --product-card-border-color: var(--color-secondary);
    --product-card-width: 344px;
`;

export const Sentinel = styled.div`
    height: 1px;
`;

export const Loading = styled.div`
    text-align: center;
    margin-top: ${({ theme }) => theme.spacing(2)};
    color: ${({ theme }) => theme.colors.primary};
`;
