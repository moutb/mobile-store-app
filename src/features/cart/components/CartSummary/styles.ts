// components/cart/styles/CartSummary.styles.ts
import { media } from '@/styles/media';
import Link from 'next/link';
import styled from 'styled-components';
import { css } from 'styled-components';

export const SummaryContainer = styled.div`
    display: flex;
    flex-direction: column;
    gap: var(--spacing-4);
    margin-top: var(--spacing-6);
    padding-bottom: var(--spacing-2);
`;

export const Actions = styled.div`
    display: flex;
    gap: var(--spacing-2);
    flex-wrap: wrap;
    width: 100%;
    justify-content: space-between;
    align-items: center;

    ${media('md')(css`
        flex-wrap: nowrap;
        justify-content: flex-end;
        gap: var(--spacing-6);
    `)}
`;

export const TotalRow = styled.div`
    display: flex;
    justify-content: space-between;
    width: 100%;
    gap: var(--spacing-4);
    font-size: var(--font-size-sm);
    font-weight: var(--font-weight-thin);

    ${media('md')(css`
        justify-content: flex-end;
    `)}
`;

export const TotalLabel = styled.span`
    color: var(--color-text-secondary);
`;

export const TotalPrice = styled.span`
    color: var(--color-text);
`;

const ActionButton = css`
    padding: var(--spacing-4) 0;
    cursor: pointer;
    font-size: var(--font-size-xs);
    line-height: var(--font-size-xs);
    font-weight: var(--font-weight-thin);
    text-transform: uppercase;
    width: calc(50% - var(--spacing-2));
    white-space: nowrap;
    text-align: center;
    margin-top: var(--spacing-4);

    ${media('md')(css`
        max-width: 260px;
        padding: var(--spacing-4) var(--spacing-6);
        font-size: var(--font-size-sm);
        line-height: var(--font-size-sm);
        margin-top: 0;
    `)}
`;

export const OutlinedButton = styled(Link)`
    ${ActionButton}
    background: none;
    color: var(--color-text);
    border-width: 0.5px;
    border-style: solid;
    border-color: var(--color-primary);

    &:hover {
        background: var(--color-secondary);
    }

    &:focus-visible {
        outline: 2px solid var(--color-primary);
        outline-offset: 2px;
    }

    ${media('md')(css`
        order: -1;
    `)}
`;

export const FilledButton = styled.button`
    ${ActionButton}
    background: var(--color-primary);
    color: var(--color-white);
    border: none;

    &:hover {
        opacity: 0.9;
    }

    &:focus-visible {
        outline: 2px solid var(--color-primary);
        outline-offset: 2px;
    }
`;
