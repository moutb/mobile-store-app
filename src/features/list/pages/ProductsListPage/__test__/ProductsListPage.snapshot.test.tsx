// __tests__/ProductListPage.snapshot.test.tsx
import { render } from '@testing-library/react';
import '@/__test__/mocks/next-navigation';
import ProductListPage from '..';
import { ThemeProvider } from 'styled-components';
import { theme } from '@/styles/theme';
import defaultProps from './default-props';
import '@/__test__/mocks/intersection-observer';

describe('<ProductListPage /> snapshot', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <ProductListPage {...defaultProps} />
            </ThemeProvider>,
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
