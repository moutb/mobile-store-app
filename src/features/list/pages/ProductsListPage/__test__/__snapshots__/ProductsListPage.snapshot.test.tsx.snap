// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductListPage /> snapshot matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  align-items: flex-start;
  justify-content: space-between;
  gap: var(--spacing-3);
}

.c1 {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  gap: 0;
}

.c2 {
  display: flex;
  align-items: center;
  border-bottom: 1px solid #000;
  width: 100%;
  position: relative;
}

.c3 {
  flex: 1;
  max-width: 100%;
  font-size: var(--font-size-lg);
  line-height: var(--font-size-lg);
  font-weight: var(--font-weight-thin);
  padding-top: var(--spacing-2);
  padding-bottom: var(--spacing-2);
  padding-right: var(--spacing-4);
  border: none;
  outline: none;
  background: transparent;
}

.c3::placeholder {
  color: var(--color-light);
  font-weight: var(--font-weight-thin);
}

.c3:disabled {
  opacity: 0.5;
}

.c6 {
  padding: var(--spacing-4);
  border: var(--product-card-border-width) solid var(--color-primary);
  position: relative;
  background: var(--color-background);
  transition: background-color 0.3s ease-in-out;
}

.c6::before {
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 0%;
  background-color: var(--color-primary);
  z-index: 0;
  transition: height 0.8s cubic-bezier(0.1,1,0.8,1);
}

.c6:hover::before {
  height: 100%;
}

.c6>* {
  position: relative;
  z-index: 1;
}

.c8 {
  width: 100%;
  aspect-ratio: 1/1;
  display: flex;
  justify-content: center;
  align-items: center;
  overflow: hidden;
}

.c9 {
  width: 100%;
  height: 100%;
  object-fit: contain;
  user-drag: none;
  -webkit-user-drag: none;
  pointer-events: auto;
}

.c7 {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  height: 100%;
  text-decoration: none;
  color: inherit;
  gap: 1rem;
  transition: color 0.3s ease-in-out;
}

.c5:hover .c7 {
  color: var(--color-white);
}

.c10 {
  display: grid;
  grid-template-columns: 1fr auto;
  gap: var(--spacing-1);
}

.c11 {
  grid-column: 1/-1;
  font-size: var(--product-card-brand-font-size);
  text-transform: uppercase;
  font-weight: var(--product-card-brand-font-weight);
}

.c12 {
  font-size: var(--product-card-name-font-size);
  text-transform: uppercase;
  font-weight: var(--product-card-name-font-weight);
  line-height: var(--product-card-name-line-height);
}

.c13 {
  font-size: var(--product-card-price-font-size);
  font-weight: var(--product-card-price-font-weight);
  line-height: var(--product-card-price-line-height);
  align-self: end;
  justify-self: end;
}

.c4 {
  display: grid;
  grid-template-columns: repeat( auto-fit,minmax(var(--product-card-width),1fr) );
  width: 100%;
  margin-top: var(--spacing-4);
  --product-card-border-width: 1px;
  --product-card-border-color: var(--color-secondary);
  --product-card-width: 344px;
}

.c4>* {
  border: var(--product-card-border-width) solid var(--color-primary);
  height: 344px;
}

.c14 {
  height: 1px;
}

<div
    class="c0"
  >
    <h1
      class="sr-only"
    >
      Products list
    </h1>
    <form
      aria-label="Search products"
      class="c1"
      role="search"
      style="width: 100%;"
    >
      <label
        class="sr-only"
        for="search"
      >
        Search
      </label>
      <div
        class="c2"
      >
        <input
          class="c3"
          id="search"
          name="search"
          placeholder="Search for a smartphone..."
          type="text"
          value=""
        />
      </div>
    </form>
    <div
      class="c4"
    >
      <article
        aria-labelledby="product-SMG-S24U"
        class="c5 c6 product--card-item"
      >
        <a
          aria-label="View details of Galaxy S24 Ultra"
          class="c7"
          href="/products/SMG-S24U"
        >
          <div
            class="c8"
          >
            <img
              alt="Picture of Galaxy S24 Ultra"
              class="c9"
              data-nimg="1"
              decoding="async"
              draggable="false"
              height="240"
              loading="lazy"
              src="/_next/image?url=http%3A%2F%2Fprueba-tecnica-api-tienda-moviles.onrender.com%2Fimages%2FSMG-S24U-titanium-violet.webp&w=640&q=75"
              srcset="/_next/image?url=http%3A%2F%2Fprueba-tecnica-api-tienda-moviles.onrender.com%2Fimages%2FSMG-S24U-titanium-violet.webp&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Fprueba-tecnica-api-tienda-moviles.onrender.com%2Fimages%2FSMG-S24U-titanium-violet.webp&w=640&q=75 2x"
              style="color: transparent;"
              width="240"
            />
          </div>
          <div
            class="c10"
          >
            <span
              class="c11"
            >
              Samsung
            </span>
            <h2
              class="c12"
              id="product-SMG-S24U"
              tabindex="-1"
            >
              Galaxy S24 Ultra
            </h2>
            <span
              class="c13"
            >
              1329 EUR
            </span>
          </div>
        </a>
      </article>
      <article
        aria-labelledby="product-SMG-A25"
        class="c5 c6 product--card-item"
      >
        <a
          aria-label="View details of Galaxy A25 5G"
          class="c7"
          href="/products/SMG-A25"
        >
          <div
            class="c8"
          >
            <img
              alt="Picture of Galaxy A25 5G"
              class="c9"
              data-nimg="1"
              decoding="async"
              draggable="false"
              height="240"
              loading="lazy"
              src="/_next/image?url=http%3A%2F%2Fprueba-tecnica-api-tienda-moviles.onrender.com%2Fimages%2FSMG-A25-negro.webp&w=640&q=75"
              srcset="/_next/image?url=http%3A%2F%2Fprueba-tecnica-api-tienda-moviles.onrender.com%2Fimages%2FSMG-A25-negro.webp&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Fprueba-tecnica-api-tienda-moviles.onrender.com%2Fimages%2FSMG-A25-negro.webp&w=640&q=75 2x"
              style="color: transparent;"
              width="240"
            />
          </div>
          <div
            class="c10"
          >
            <span
              class="c11"
            >
              Samsung
            </span>
            <h2
              class="c12"
              id="product-SMG-A25"
              tabindex="-1"
            >
              Galaxy A25 5G
            </h2>
            <span
              class="c13"
            >
              239 EUR
            </span>
          </div>
        </a>
      </article>
      <div
        aria-hidden="true"
        class="c14"
        data-testid="infinite-scroll-sentinel"
        role="presentation"
      />
    </div>
  </div>
</DocumentFragment>
`;
